### ISSUE TITLE
Disable new signups

### ISSUE THREAD

**Issue by karkazzul** [pglombardo/PasswordPusher#378]
Hi,

When, i have created my users, i want disable new singups. 

Currently, i don't think that is possible ?

Thk

**Comment by pglombardo** [comment]
There are couple ways to do this:

1. If you are using a proxy in front, block paths that end with `/users/sign_up` (because there is a lang code such as `/en/users/sign_up`)
2. Alternatively, you could remove `new create` from https://github.com/pglombardo/PasswordPusher/blob/master/config/routes.rb#L12

For #2,  you would then to mount the modified `routes.rb` file into your container (assuming you are using containers) with a command line flag such as `--mount type=bind,source=/path/routes.rb,target=/opt/PasswordPusher/config/routes.rb` 

I might be able to add an environment variable to disable this in the future but this is the first such request I've gotten for this.

Does one of these options seem feasible to you?

**Comment by karkazzul** [comment]
Thanks for your answer. I use traefik like a reverse proxy for docker. I think that an middleware for blocking the path `/users/sign_up` is a good idea. 

I'll let you know if it works

**Comment by neander** [comment]
I would love this functionality as well. It would be useful for usage by a small group of friends with access to my self-hosted server.

**Comment by pglombardo** [comment]
Considering the number of requests for this: #422 

Set `PWP__DISABLE_SIGNUPS=true` (or equivalent in `settings.xml`) and no more user account creations are allowed.  The links are removed and backend functionality disabled. 

I'll do a bit more testing and see if I can release it this week.

**Comment by pglombardo** [comment]
This feature has been added and released in [version 1.17.5](https://github.com/pglombardo/PasswordPusher/releases/tag/v1.17.5).

Documentation for the environment variable can be found at the [end of this table](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#enabling-logins).

tldr: `PWP__DISABLE_SIGNUPS=true`

Thanks all for the request, input and feedback!

**Comment by neander** [comment]
Impressive speed of development. Thank you!
